crates/next-api/src/server_actions.rs

```diff
@@ -129,10 +136,11 @@
         )?;
     }
 
-    let path = project_path.join(format!(".next-internal/server/app{page_name}/actions.js").into());
     let file = File::from(contents.build());
     let source = VirtualSource::new_with_ident(
-        AssetIdent::from_path(path).with_modifier(server_actions_loader_modifier()),
+        rsc_entry
+            .ident()
+            .with_modifier(server_actions_loader_modifier()),
```

Comment:
It might need to be a path that ends with `.js` or do we have a special case for ReferenceType::Internal to resolve as ecmascript?

Reply:
Realistically, that should always be the case already anyway. But I've added it anyway